zerver/lib/rate_limiter.py

```diff
@@ -591,6 +593,9 @@ def rate_limit_ip(request: HttpRequest, ip_addr: str, domain: str) -> None:
 
 
 def rate_limit_request_by_ip(request: HttpRequest, domain: str) -> None:
+    if not should_rate_limit(request):
+        return
```

Comment:
It's kind of akward that we now have the inconsistency that `rate_limit_user` checks ` should_rate_limit(request)`, but the analogical `rate_limit_ip` just rate limits unconditionally because the check is here. I think checking here might make sense to skip the `TOR` logic below if possible, but maybe we should put a duplicate 
```
    if not should_rate_limit(request):
        return
```

in `rate_limit_ip` just for consistency? :thinking:  @timabbott  Thoughts here?

Reply:
I think it might be better to just remove `rate_limit_ip` because `rate_limit_request_ip` is the only caller. We only need to update the mocked test case.

Reply:
Ohh yeah, that sounds like the right idea to me